src/connectivity/mod.rs

```diff
@@ -981,27 +981,54 @@ pub fn graph_all_simple_paths(
             "The input index for 'from' is not a valid node index",
         ));
     }
-    let to_index = NodeIndex::new(to);
-    if !graph.graph.contains_node(to_index) {
-        return Err(InvalidNode::new_err(
-            "The input index for 'to' is not a valid node index",
-        ));
-    }
     let min_intermediate_nodes: usize = match min_depth {
         Some(0) | None => 0,
         Some(depth) => depth - 2,
     };
     let cutoff_petgraph: Option<usize> = cutoff.map(|depth| depth - 2);
-    let result: Vec<Vec<usize>> = algo::all_simple_paths::<Vec<_>, _, foldhash::fast::RandomState>(
+
+    if let Ok(to) = to.extract::<usize>() {
+        let to_index = NodeIndex::new(to);
+        if !graph.graph.contains_node(to_index) {
+            return Err(InvalidNode::new_err(
+                "The input index for 'to' is not a valid node index",
+            ));
+        }
+
+        let result: Vec<Vec<usize>> =
+            algo::all_simple_paths::<Vec<_>, _, foldhash::fast::RandomState>(
+                &graph.graph,
+                from_index,
+                to_index,
+                min_intermediate_nodes,
+                cutoff_petgraph,
+            )
+            .map(|v: Vec<NodeIndex>| v.into_iter().map(|i| i.index()).collect())
+            .collect();
+        return Ok(result);
+    }
+
+    let mut target_set: HashSet<NodeIndex> = HashSet::new();
```

Comment:
I think most users will pass in Python lists and expect output to be consistent. Use https://docs.rs/indexmap/latest/indexmap/set/struct.IndexSet.html with `foldhash` as the hasher.